staging/src/k8s.io/legacy-cloud-providers/azure/azure.go

```diff
@@ -819,6 +819,11 @@ func (az *Cloud) updateNodeCaches(prevNode, newNode *v1.Node) {
 		if _, hasExcludeBalancerLabel := prevNode.ObjectMeta.Labels[v1.LabelNodeExcludeBalancers]; hasExcludeBalancerLabel {
 			az.excludeLoadBalancerNodes.Delete(prevNode.ObjectMeta.Name)
 		}
+
+		// If not in ready state, remove from excludeLoadBalancerNodes cache.
+		if !isNodeReady(prevNode) {
```

Comment:
With https://github.com/kubernetes/kubernetes/pull/108284/files#r813266818.  We make an explicit either/or below.  Sorry I didn't notice prior